spec/models/provider_spec.rb

```diff
@@ -61,4 +61,38 @@
       expect(tenant.providers).to include(provider)
     end
   end
+
+  context "#orchestrate_destroy" do
+    let(:manager) { FactoryGirl.create(:ext_management_system) }
+
+    it "destroy when has no managers" do
+      expect(provider).to receive(:destroy)
+      provider.destroy_queue
+    end
+
+    it "call orchestrate_destroy its managers first" do
+      expect(manager).to receive(:enabled) { true }
+      provider.managers = [manager]
+      expect(manager).to receive(:destroy_queue)
+      expect(provider).not_to receive(:destroy)
+      provider.destroy_queue
+    end
+
+    it "doesn't orchestrate_destroy its managers when they are disabled" do
+      expect(manager).to receive(:enabled) { false }
+      provider.managers = [manager]
+      expect(manager).not_to receive(:destroy_queue)
+      expect(provider).not_to receive(:destroy)
+      provider.destroy_queue
+    end
+
+    it "queues itself for orchestrate_destroy when managers exists" do
+      allow(Time).to receive(:now).and_return(Time.zone.now)
+      provider.managers = [manager]
+      expect(manager).to receive(:destroy_queue)
+      expect(provider).not_to receive(:destroy)
+      expect(described_class).to receive(:_queue_task).with(:destroy_queue, provider.id.to_miq_a, 15.seconds.from_now)
```

Comment:
Typically we `expect(MiqQueue.find_by(:class_name =>"X", instance_id => n, ...).to have_attributes(…)`

Reply:
I'd like to keep these tests remains in the scope of this `destroy_queue`.  Testing of resulting effect in `MiqQueue` would belong to those for `_queue_task` method (or the subsequently called `MiqQueue.put`.

Reply:
@Fryguy what do you think?